consensus/state.go

```diff
@@ -2844,3 +2849,15 @@ func (cs *State) syncData() {
 		}
 	}
 }
+
+func (cs *State) lockAll() {
+	cs.mtx.Lock()
+	cs.stateMtx.Lock()
+	cs.rsMtx.Lock()
+}
```

Comment:
hum, I think if we move the state and rs mutexes to their structs, this could be simplifed

Reply:
I see, let's leave it this way then, thanks 🙏